packages/xchain-util/src/asset.test.ts

```diff
@@ -46,6 +90,50 @@ describe('asset', () => {
       expect(amount.type).toEqual(Denomination.BASE)
       expect(amount.amount()).toEqual(bn('10'))
     })
+    it('should be able to add a value', () => {
+      const amount = baseAmount(10).plus(100)
+      expect(amount.type).toEqual(Denomination.BASE)
+      expect(amount.amount()).toEqual(bn('110'))
+    })
+    it('should be able to sub a value', () => {
+      const amount = baseAmount(10).minus(5)
+      expect(amount.type).toEqual(Denomination.BASE)
+      expect(amount.amount()).toEqual(bn('5'))
+    })
+    it('should be able to mul a value', () => {
+      const amount = baseAmount(10).times(5)
+      expect(amount.type).toEqual(Denomination.BASE)
+      expect(amount.amount()).toEqual(bn('50'))
+    })
+    it('should be able to div a value', () => {
+      const amount = baseAmount(10).div(5)
+      expect(amount.type).toEqual(Denomination.BASE)
+      expect(amount.amount()).toEqual(bn('2'))
+    })
+    it('should be able to add BaseAmount', () => {
+      const amount = baseAmount(10, 18).plus(baseAmount(10, 10))
+      expect(amount.type).toEqual(Denomination.BASE)
+      expect(amount.amount()).toEqual(bn('20'))
+      expect(amount.decimal).toEqual(18)
+    })
+    it('should be able to set decimal', () => {
+      const amount = baseAmount(10, 18).plus(baseAmount(10, 10), 12)
+      expect(amount.type).toEqual(Denomination.BASE)
+      expect(amount.amount()).toEqual(bn('20'))
+      expect(amount.decimal).toEqual(12)
+    })
+    it('should be able to compare BaseAmount', () => {
+      expect(baseAmount(10).lt(baseAmount(11))).toEqual(true)
+      expect(baseAmount(10).lte(baseAmount(10))).toEqual(true)
+      expect(baseAmount(10).gt(baseAmount(9))).toEqual(true)
+      expect(baseAmount(10).gte(baseAmount(10))).toEqual(true)
+      expect(baseAmount(10).eq(baseAmount(10))).toEqual(true)
+      expect(baseAmount(10).lt(11)).toEqual(true)
+      expect(baseAmount(10).lte(10)).toEqual(true)
+      expect(baseAmount(10).gt(9)).toEqual(true)
+      expect(baseAmount(10).gte(10)).toEqual(true)
+      expect(baseAmount(10).eq(10)).toEqual(true)
+    })
```

Comment:
Pls extract tests for `lt`, `lte`, `gt`, `gte` into different testcases to test one compare function (`lt` or` `lte` etc) only.